### ISSUE TITLE
States should not be able to have same name

### ISSUE THREAD

**Issue by Meorge** [AutomatonBuilderGUI/AutomatonBuilderGUI#37]

**Comment by ryan-ball-360** [comment]
an error message has been created that displays a warning when the user changes a label name to one that is already present in the finite automata

**Comment by Meorge** [comment]
Being addressed with #43

**Comment by Meorge** [comment]
As stated in the pull request comments, it is possible to get the message to disappear without resolving the state naming problem.

Steps to reproduce:
1. Create three states: q0, q1, and q2.
2. Rename q2 to q0. The error message should appear.
3. Delete q1. The error message should disappear, but the two remaining states still have the same label.

I inserted the following line of code into the `StateManager.areAllLabelsUnique()` method:
```ts
console.log(`Labels are`, labels, 'unique labels are', uniqueLabels);
```

The results I'm getting are quite strange:
![CleanShot 2024-03-04 at 16 55 08@2x](https://github.com/AutomatonBuilderGUI/AutomatonBuilderGUI/assets/9957987/0124c699-4a2b-4d6b-9196-cfbf26f05d3c)

For some reason, it looks like it's... internally deleting all of the states, *except* for the state we just deleted.

Looking further through the code, I think I found where the bug is, and I'll open a new issue for it!
